File: benchmarks/file_io/readwrite_10mb.py
import time, os, tempfile
data = "x" * (10 * 1024 * 1024)
tmp = tempfile.mktemp(suffix='.bench')
start = time.monotonic_ns()
with open(tmp, 'w') as f:
    f.write(data)
with open(tmp, 'r') as f:
    _ = f.read()
end = time.monotonic_ns()
os.unlink(tmp)
print((end - start) // 1000)

File: benchmarks/file_io/readwrite_10mb.rb
require 'tempfile'
data = "x" * (10 * 1024 * 1024)
tmp = Tempfile.new('bench')
path = tmp.path
tmp.close
start = Process.clock_gettime(Process::CLOCK_MONOTONIC, :microsecond)
File.write(path, data)
File.read(path)
finish = Process.clock_gettime(Process::CLOCK_MONOTONIC, :microsecond)
File.delete(path) rescue nil
puts(finish - start)
